### ISSUE TITLE
Local mouse with dynamic icon state

### ISSUE THREAD

**Issue by rogtheo** [moonlight-stream/moonlight-qt#1016]
I use RDP, VNC and Parsec for remote desktop and moonlight/sunshine for game streaming. The only thing stopping me from completely migrating to moonlight is the remote mouse. Moonlight gives the best video performance but feels worse than even RDP on desktop because of seeing the remote mouse that adds some latency to the movement.

I've been using the "hide remote mouse" (doesnt always work) and "show local mouse" hotkeys and when seeing the local mouse it feels like I'm on my actual PC! The only issue is that the cursor is always a pointer since it can't recognise what I hover on the remote machine and it also turns to a black colour pointer when I'm on a Raspberry Pi or android(since that is the OS cursor colour).

Parsec has this implemented by default and they support it in game as well so you never need to swap modes.
How easy is this to implement? It seems to be the standard with other software like VNC, RDP and Parsec.

**Comment by rodalpho** [comment]
I do the exact same thing. Would love to get rid of parsec and only use moonlight/sunshine for everything.

**Comment by cgutman** [comment]
> How easy is this to implement? 

Not particularly easy (there are many cursor formats and many renderers to update), but it's on the roadmap.

**Comment by eric-gitta-moore** [comment]
Yes, the cursor is part of the user experience. I have tested several clients, one of which is worth mentioning the mac as the client, and it seems that this feature does not work (Ctrl+Alt+Shift+C - Toggle local cursor display in remote desktop mouse mode)

**Comment by cryonox** [comment]
hello I have found a workaround to this problem by using an external program. hopefully moonlight and sunchine will implemt similar solution natively. https://github.com/cryonox/moonlight_mousesync

**Comment by rogtheo** [comment]
> hello I have found a workaround to this problem by using an external program. hopefully moonlight and sunchine will implemt similar solution natively. https://github.com/cryonox/moonlight_mousesync

You are a life saver!!

**Comment by eric-gitta-moore** [comment]
> Yes, the cursor is part of the user experience. I have tested several clients, one of which is worth mentioning the mac as the client, and it seems that this feature does not work (Ctrl+Alt+Shift+C - Toggle local cursor display in remote desktop mouse mode)

Now, if I give you feedback, it's magic. Now my shortcut Ctrl+Alt+Shift+C is working

**Comment by Enduriel** [comment]
I really hope this gets resolved properly in the main repo, I am trying to use moonlight over wan and therefore do not want to have a permanently open port running for the script. For me this is also the only thing keeping me from fully switching over from parsec. Hope this gets prioritized and thanks for the amazing work everyone has done on the project.

**Comment by uncleungie** [comment]
This is also the final issue before I switch from Parsec. Local cursor rendering is a must.
Otherwise, thanks for all the hard work on this amazing project.

**Comment by jorys-paulin** [comment]
> It might only be a small setting called "do not hide the cursor while in connection", but it will truly helps a lot when using moonlight windows as a remote desktop software, it is so perfect for local use.

So do you mean a setting for `Ctrl+Alt+Shift+C - Toggle local cursor display in remote desktop mouse mode`, like there is for the performance overlay for example?

**Comment by LineCatOvO** [comment]
> So do you mean a setting for `Ctrl+Alt+Shift+C - Toggle local cursor display in remote desktop mouse mode`, like there is for the performance overlay for example?

I dont know if this feature is already implemented? But I am using moonlight windows 6.1.0 and this shortcut is not working, but at least ctrl+alt+shift+n can remove the host cursor in client view. If you are talking about that plugin [https://github.com/cryonox/moonlight_mousesync](url) then installing another nearly independent one is a little complex...

**Comment by fisherman10** [comment]
Any updates on this so far?
